install/kubernetes/cilium/values.yaml

```diff
@@ -1643,3 +1643,8 @@ clustermesh:
 externalWorkloads:
   # -- Enable support for external workloads, such as VMs (false by default).
   enabled: false
+
+# -- Configure cgroup related configuration
+cgroup:
+  # -- Configure cgroup root where cgroup2 filesystem is mounted on the host (see also: `cgroup.autoMount`)
+  hostRoot: /run/cilium/cgroupv2
```

Comment:
shouldn't this be by default `/sys/fs/cgroup`? The comment is a little confusing. Looking at the helm chart (install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml line 281) it looks that this is used to define the path where cgroup-root is mounted inside the container.

edit since I was reviewing the PR per commit basis: It also seems that with `cgroup.autoMount=false` Cilium will not mount the cgroup v2 correctly under `/sys/fs/cgroup` as there is no reference of it anywhere.

Reply:
See this thread - https://github.com/cilium/cilium/pull/16259#discussion_r656486823.  😄 

Also, the reason I decided to rename the option is because this is user-facing. Users shouldn't care where we mount the fs inside the agent pod.

>It also seems that with cgroup.autoMount=false Cilium will not mount the cgroup v2 correctly under /sys/fs/cgroup

See this comment - 

```
# If users disable `autoMount`, it's expected that users have mounted
# cgroup2 filesystem at the specified `cgroup.hostRoot` 
```

Reply:
> See this thread - [#16259 (comment)](https://github.com/cilium/cilium/pull/16259#discussion_r656486823). 
> 
> @kkourt suggested to rename the option. This makes sense since it is user-facing. Users shouldn't care where we mount the fs inside the agent pod.
> 
> > It also seems that with cgroup.autoMount=false Cilium will not mount the cgroup v2 correctly under /sys/fs/cgroup
> 
> See this comment -
> 
> ```
> # If users disable `autoMount`, it's expected that users have mounted
> # cgroup2 filesystem at the specified `cgroup.hostRoot` 
> ```

But why isn't the default `/sys/fs/cgroup` and `/sys/fs/cgroup/unified` for gke? This would work as fine regardless if `autoMount` is true or false. However if we have `/run/cilium/cgroupv2` set as default, this will not work by default for users running with `autoMount` as false, which we can avoid it by setting the default to `/sys/fs/cgroup` for `/sys/fs/cgroup/unified` for gke.

Reply:
Defaulting to mounting the cgroupfs to `/sys/fs/cgroup` on the host sounds fine to me, I doubt that there's much problem there - it's the canonical path so it's not like we'd get some other filesystem mounted there. If it's already mounted, we can detect that and avoid mounting it. If it's not yet mounted, then I don't think anyone would complain if we mounted it. Every mount of the cgroupv2 FS points to the same underlying cgroupfs implementation so there's nothing special about us mounting it there vs any other software mounting it there, so from that reasoning I don't think there's a compatibility issue or anything.

Reply:
@aditighag, while I was discussing this topic with Joe we are a little bit confused about one thing: We thought the whole point of `autoMount=true` was to auto detect the cgroup root mount point, so why are we using `/var/run/cilium/cgroupv2` in `autoMount=true`? Or is a misunderstanding from our part?

Reply:
@joestringer To be clear, I haven't changed the default path. You explained the reasoning in the sig-datapath meeting why we chose to mount cgroup2 fs at a custom location. 

@aanm It's auto mount, where the daemonset will automatically **mount** cgroup2 fs.

Reply:
>@aanm It's auto mount, where the daemonset will automatically mount cgroup2 fs.

oh and that is because the init container will mount the cgroup2 fs in that directory and then cilium container can use it?

Reply:
Evidently past me was thinking about this more deeply than current me :sweat_smile: (Looking at the notes, I don't think we captured this nuance) . I think I recall the reasoning, it was something along the lines of, if the host isn't already mounting cgroupfs by itself then maybe /sys/fs/cgroup doesn't exist as a directory to mount to (because otherwise the OS would mount the FS there right?) and then maybe it's not our role to create that directory. Then it's better to just mount on `/var/run/cilium/cgroupv2` where we know that Cilium controls the directory and no-one should interfere with it.

OK, then I think this default is fine as-is.